aries_vcx/Cargo.toml

```diff
@@ -15,31 +15,13 @@ default = ["vdrtools"]
 vdrtools = ["aries_vcx_core/vdrtools"]
 # Feature flag to include the 'modular library' dependencies (vdrtools alternatives; indy-vdr, indy-credx)
 modular_libs = ["aries_vcx_core/modular_libs"]
-########################## TEST FLAGS ###############################
-# Feature flag for base requirments for running tests underneath the `test_utils` flag (most tests).
-# Both `vdrtools` and `modular_libs` are transitively enabled, so they can be used for tests (including interop tests).
-test_utils = [ "messages/test_utils", "aries_vcx_core/test_utils", "vdrtools", "modular_libs" ]
-# Feature flag for categorising tests which require a ledger to be running
-pool_tests = [ "test_utils" ]
-# Feature flag for categorising tests which require a ledger and messaging agency to be running
-agency_pool_tests = [ "test_utils" ]
-# Feature flag for categorising general tests (usually unit tests)
-general_test = [ "test_utils", "messages/general_test" ]
-# Feature flag to enable experimental tests of a mysql wallet
-mysql_test = ["test_utils" ]
-# Feature flag used to prompt some tests to run using the `modular_libs` dependencies instead of `vdrtools`
-modular_libs_tests = ["test_utils", "tokio/rt", "tokio/macros"]
-########################## OTHER FLAGS ##############################
-# Feature flag to make compiler warnings into failures
-fatal_warnings = []
-# Feature flag to log some extra warnings if enabled
-warnlog_fetched_messages = []
+
 
 [dependencies]
 agency_client = { path = "../agency_client" }
 messages = { path  = "../messages" }
-# default-features as false, since we transitively choose features via `vdrtools` and `modular_libs` features
-aries_vcx_core = { path  = "../aries_vcx_core", default-features = false }
+diddoc = { path = "../diddoc" }
+aries_vcx_core = { path  = "../aries_vcx_core" }
```

Comment:
this is a regression on a regression fix 😅.. removing `default-features = false` from the `aries_vcx_core` dependency forces all `aries_vcx` consumers to have a dependency on `vdrtools`, even when they have specified they don't want that dependency.

For some (including myself), this is a fatal regression, as `vdrtools` is full of old dependencies that can cause tremendous dependency conflicts (e.g. with aries_askar and other more modern crates).

Trying not to nitpick, but I consider this a fatal regression - was something about `default-features = false` causing you issues?? if not, would be good if you could add it back

Reply:
It probably was if I removed it, but I'll look into it. Good catch

Reply:
thnx. Lets see how the CI reacts to that...

Reply:
Hmm. I believe a quick fix for the issue in the CI: [check-aries-vcx-feature-variants](https://github.com/hyperledger/aries-vcx/actions/runs/4806298131/jobs/8553648204#logs), is to add the `vdrtools` feature flag above the `test_utils` and `devsetup` modules:

```
#[cfg(feature = "vdrtools")]
pub mod devsetup;
```

```
#[cfg(feature = "vdrtools")]
pub mod test_utils;
```

Reply:
@bobozaur  your recent fix may cause issues in libvcx, since it depends on test_utils and/or devsetup i believe... (which is not the best IMO)..

Adding `[cfg(feature = 'vdrtools')]` may fix the problem with the CI step you're trying to fix.. unsure if it will break tests tho..

Reply:
hahhahah yep, it's a bit tangled. I think removing the testing flags is a good first step though.

Reply:
A good first step to hell

Reply:
Went at this differently, because of the hundreds of errors this generated.
Removed the `default` from `aries_vcx_core` instead. This means that neither `vdrtools` nor `modular_libs` are selected implicitly, so are controlled instead from `aries_vcx`.

The benefit is that it does not require changes in `aries_vcx` where every line change causes an avalanche of errors. We really need to put effort, a lot of it, into getting this into a better shape. 

I've had more fun getting punched in the face. Hopefully the CI will run okay.

Reply:
yup, that approach makes sense. 

It's close, but compiling aries_vcx with `--features modular_libs --no-default-features` and `--no-default-features` won't work bcus test_utils and devsetup assumes vdrtools is on. 

I'm thinking that the quick fix is to add the `[cfg(feature = 'vdrtools')]` for test_utils and devsetup

Reply:
it's very unfortunate that `tests/` and `libvcx` depend on functionality within `test_utils` and `devsetup` :(

Reply:
There are things in `devsetup` that depend on `modular_libs`. We can't just add `vdrtools` to it...

Reply:
Not sure if that matters; currently tests must be ran with either; just the `vdrtools` flag (default), or both the `vdrtools` and `modular_libs` flag (default  + modular_libs). Never is it just `modular_libs`.

`devsetup` and `test_utils` are only used by tests, and those tests are only ran if atleast `vdrtools` is enabled.

It's a bit sketchy, but it's a quick fix to get you thru it I believe.

With this fix locally, I've been able to run a good amount of integration tests (both vdrtools and modular_libs), and all the `cargo check` variants pass (since these cargo checks are not checking any `cfg(test)` code)

Reply:
Hm... Let me try.

Reply:
It looks like it works. Let's see the CI. I don't understand why they're not exclusive and at this point I am too afraid to ask.

Reply:
The main reason is that a lot of `Profile` functionality (issuer and wallet) has not been implemented with `modular_libs` yet. So having pure 'modular_libs' end-to-end tests aren't possible, which means there's a bit of weird hybrid tests and all the end to end tests still depend on vdrtools a little bit... 

Once we reach pure `modular_libs` implementation, they can be exclusive, and a lot of the conditional mess can be cleaned up

Reply:
All ran now!